src/Castle.Core/DynamicProxy/ProxyGenerationOptions.cs

```diff
@@ -204,7 +208,53 @@ public override int GetHashCode()
 			result = 29*result + (Selector != null ? 1 : 0);
 			result = 29*result + MixinData.GetHashCode();
 			result = 29*result + (BaseTypeForInterfaceProxy != null ? BaseTypeForInterfaceProxy.GetHashCode() : 0);
+		    result = 29*result + GetContentsHashCode(AdditionalAttributes);
 			return result;
 		}
+
+        private static int GetContentsHashCode<T>(IList<T> list)
+        {
+            var result = 0;
+            for (var i = 0; i < list.Count; i++)
+            {
+                if (list[i] != null)
+                {
+                    // simply XOR since order does not matter
+                    result ^= list[i].GetHashCode();
+                }
+            }
+
+            return result;
+        }
+
+        private static bool ListsEquivalent<T>(IList<T> listA, IList<T> listB)
+        {
+            if (listA.Count != listB.Count)
+            {
+                return false;
+            }
+
+            // order is not important, just make sure that each entry in A is also found in B

```

Comment:
I worry about this implementation. If we're guaranteed to have only one instance in each list that's equal to one instance in the other list, we're okay, but if duplicates are allowed (and I'll say right now that I don't follow well enough to know if duplicates are possible or not), then the lists `<a, a, b>` and `<a, a, b>` will compare as equivalent, even though they are not. Perhaps more damning, the hash codes would be _not_ be equal.